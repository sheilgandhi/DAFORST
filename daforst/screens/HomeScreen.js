import React, { useEffect, useLayoutEffect } from 'react';
import {
  Image,
  StyleSheet,
  Text,
  View,
  SafeAreaView,
  Pressable,
  useWindowDimensions,
} from 'react-native';

import { COLORS } from '../colors';

const image = require('../assets/daforst_tree.png');

const HomeScreen = ({ navigation }) => {
  const { height, width } = useWindowDimensions();

  useLayoutEffect(() => {
    navigation.setOptions({
      title: 'DAFORST',
      headerShown: false,
    });
  }, [navigation]);

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.cta}>
        <Image style={styles.image(height)} source={image} />
        <Text style={styles.lineOne}>Hey! Welcome</Text>
        <Text style={styles.lineTwo}>
          DAFORST is a decentralised platform which helps provide data
          sovereignty to communities.
        </Text>
      </View>
      <View style={styles.buttons}>
        <Pressable
          onPress={() => navigation.navigate('Upload')}
          style={[
            styles.button,
            { backgroundColor: COLORS.green, width: width - 32 },
          ]}>
          <Text style={[styles.buttonText, { color: COLORS.white }]}>
            Upload
          </Text>
        </Pressable>
        <Pressable
          onPress={() => navigation.navigate('Assets')}
          style={[
            styles.button,
            { backgroundColor: COLORS.white, width: width - 32 },
          ]}>
          <Text style={[styles.buttonText, { color: COLORS.black }]}>
            Assets
          </Text>
        </Pressable>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
  },
  cta: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-evenly',
  },
  image: height => ({
    resizeMode: 'center',
    height: height / 2,
  }),
  lineOne: {
    fontWeight: 'bold',
    color: COLORS.black,
    fontSize: 28,
  },
  lineTwo: {
    marginHorizontal: 32,
    color: COLORS.black,
    textAlign: 'center',
  },
  buttons: {
    justifyContent: 'space-evenly',
    alignItems: 'center',
    paddingBottom: 10,
  },
  button: {
    height: 50,
    borderRadius: 96,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
    borderColor: COLORS.green,
    borderWidth: 3,
  },
  buttonText: {
    fontSize: 18,
    textAlign: 'center',
  },
});

export default HomeScreen;
