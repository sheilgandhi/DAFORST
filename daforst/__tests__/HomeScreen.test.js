import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../screens/HomeScreen';
import {
  fireEvent,
  render,
  waitFor,
  cleanup,
  screen,
} from '@testing-library/react-native';

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  setOptions: jest.fn(),
};

describe('HomeScreen', () => {
  beforeEach(() => {
    render(<HomeScreen navigation={navigation} />);
  });

  afterEach(() => {
    cleanup();
  });

  test('renders correctly', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('validate call to action text', () => {
    const cta = screen.getByTestId('cta');
    expect(cta.props.children).toEqual(
      'DAFORST is a decentralised platform which helps provide data sovereignty to communities.',
    );
  });

  test('validate upload button fires', async () => {
    const pressable = screen.getByTestId('uploadButton');
    expect(pressable).toBeTruthy(); // exists?
    fireEvent.press(pressable);
    await waitFor(() => {
      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  test('validate assets button fires', async () => {
    const pressable = screen.getByTestId('assetsButton');
    expect(pressable).toBeTruthy(); // exists?
    fireEvent.press(pressable);
    await waitFor(() => {
      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
