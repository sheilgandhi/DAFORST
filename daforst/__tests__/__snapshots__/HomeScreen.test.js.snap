// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-evenly",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/daforst_tree.png",
        }
      }
      style={
        Object {
          "height": 667,
          "resizeMode": "center",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 28,
          "fontWeight": "bold",
        }
      }
    >
      Hey! Welcome
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "marginHorizontal": 32,
          "textAlign": "center",
        }
      }
      testID="cta"
    >
      DAFORST is a decentralised platform which helps provide data sovereignty to communities.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "space-evenly",
        "paddingBottom": 10,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#55897B",
            "borderRadius": 96,
            "borderWidth": 3,
            "height": 50,
            "justifyContent": "center",
            "margin": 10,
          },
          Object {
            "backgroundColor": "#55897B",
            "width": 718,
          },
        ]
      }
      testID="uploadButton"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Upload
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#55897B",
            "borderRadius": 96,
            "borderWidth": 3,
            "height": 50,
            "justifyContent": "center",
            "margin": 10,
          },
          Object {
            "backgroundColor": "#fff",
            "width": 718,
          },
        ]
      }
      testID="assetsButton"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
              "textAlign": "center",
            },
            Object {
              "color": "#000",
            },
          ]
        }
      >
        Assets
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
